centreon/www/include/views/graphs/generateGraphs/generateImage.php

```diff
@@ -176,43 +176,45 @@
 $contactId = $row['contact_id'];
 
 if (!$isAdmin) {
+
     $acl = new CentreonACL($contactId, $isAdmin);
-    $dbstorage = new CentreonDB('centstorage');
 
-    $aclGroups = $acl->getAccessGroupsString();
-    $sql = "SELECT host_id, service_id FROM index_data WHERE id = :index_data_id";
-    $statement = $dbstorage->prepare($sql);
-    $statement->bindValue(':index_data_id', (int) $index, \PDO::PARAM_INT);
-    $statement->execute();
-    if (!$statement->rowCount()) {
-        die('Graph not found');
-    }
-    $row = $statement->fetch(\PDO::FETCH_ASSOC);
-    unset($statement);
-    $hostId = $row['host_id'];
-    $serviceId = $row['service_id'];
-    $aclGroupsExploded = explode(',', $aclGroups);
-    if (empty($aclGroupsExploded)) {
+    if (empty($acl->getAccessGroups())) {
         throw new \Exception('Access denied');
     }
 
-    $aclGroupsQueryBinds = [];
-    foreach ($aclGroupsExploded as $key => $value) {
-        $aclGroupsQueryBinds[':acl_group_' . $key] = str_replace("'","",$value);
-    }
-    $aclGroupBinds = implode(',', array_keys($aclGroupsQueryBinds));
-    $sql = "SELECT service_id FROM centreon_acl WHERE host_id = :host_id AND service_id = :service_id
-        AND group_id IN ($aclGroupBinds)";
-    $statement = $pearDBO->prepare($sql);
-    $statement->bindValue(':host_id', (int) $hostId, \PDO::PARAM_INT);
-    $statement->bindValue(':service_id', (int) $serviceId, \PDO::PARAM_INT);
-    foreach ($aclGroupsQueryBinds as $key => $value) {
-        $statement->bindValue($key, (int) $value, \PDO::PARAM_INT);
-    }
-    $statement->execute();
-    if (!$statement->rowCount()) {
+    $aclGroups = implode(',', array_keys($acl->getAccessGroups()));
+
+    try {
+        $dbstorage = new CentreonDB('centstorage');
```

Comment:
I know you just scope this into a try catch and bind value, but to improve this, there is no need to instanciate $dbstorage. L.112
```
$pearDBO = new CentreonDB("centstorage");
```

pearDBO is already an instance of CentreonDB for the real time database there is no real reason to use both dbstorage and pearDBO in this method

Reply:
fixed